### ISSUE TITLE
Allow for configuring default fields on "New Contact"

### ISSUE THREAD

**Issue by hrehfeld** [nextcloud/contacts#1951]
Every time I add a new contact, I need to manually add the field: "detailed name", because i ALWAYS want to specify forename/surname/title/nickname in the appropriate fields. The ui there is very annoying and unforgiving.

Let the user/admin configure the default fields for contacts. Just a plain list of check boxes in the settings.

1.Make forename/surname the default title fields (based on locale if you're worried about "weird" names) instead of the weird concatenated name field (shown in the header of the contact)
2. Show all fields in a sensible order is also a way (my phone apps always show it that way, sometimes with details hidden in a one-click drawer -- works great).
3. A better UI for adding fields, or multiple types of "new accounts". Much more complicated than the proposed solution.

**Comment by apoeteo** [comment]
I also ran into issues with this while testing/onboarding to Nextcloud.

1. I create a contact in Nextcloud (NC) - using the default FullName field
2. Sync it to mobile (M) - works OK
3. Edit it on M
4. Sync it back to NC - works OK
5. Edit it on NC
6. Try to sync it to M - **doesn't work**

After a few hours of troubleshooting I realized that while I edit the contact on M, the FullName field will get split into FirstName and LastName.

After point 4, the detailed inspection in NC shows that the "Detailed name" field group has been added to the contact.
If I change the values in the "Detailed name" in NC and sync to M, works fine.
But the default NC FullName field won't update with the new values of FirstName and LastName, so before saving the change I also have to delete the value of FullName.

:/

**Comment by jneubert** [comment]
+1 for the proposed solution. (For me, Notes and Website are filled in oftenly.)

**Comment by tinokorth** [comment]
+1

**Comment by AKDub** [comment]
+1

**Comment by lexo-mfleuti** [comment]
+1

The way this is implemented in NC is completely wrong in my opinion. The `Detailed Name` fields must be the primary fields. The field `Name` can be auto-filled (auto-suggested) after entering Firstname + Lastname (when clicking into the field - just as it is handled in MS Outlook). A normal user will not handle contact entering correctly the way it is done now and the current solution causes all sorts of problems with various clients cutting off information entered in this field. For example people will start entering text in brackets and this will cause various issues.

Please change this behaviour. The `Detailed Name` fields must be always visible on top as default fields and the name field is just one of these fields.

I quickly created a draft on how I imagine it to be. Hope this helps:

<img width="695" height="563" alt="Image" src="https://github.com/user-attachments/assets/b60f573e-c6b7-4967-8b97-e4845f4cfcf1" />

**Comment by Peque** [comment]
I'd like to emphasize the importance of this being implemented for the user (not the admin). If you are not the owner of the instance, that would mean users have no real option to customize this. User-level fixes the issue for everyone, but admin-level does not (only fixes it for admins running their own instance). 😊

**Comment by Peque** [comment]
Possible duplicates (indicating further interest on this): #3385, #3615, #4537

**Comment by SebastianKrupinski** [comment]
@Peque thank you for linking the other tickets!

There indeed is a lot of interest in this, I will bump this up to design / product management... to see if we can get it put on the schedule.

**Comment by lexo-mfleuti** [comment]
@SebastianKrupinski @Peque Maybe my [design draft](https://github.com/nextcloud/contacts/issues/1951#issuecomment-3106745759) helps ;)

**Comment by SebastianKrupinski** [comment]
Transferred up votes from: 

#3385 - +5
#3615 - +5
#4537 - +1

**Comment by SebastianKrupinski** [comment]
@lexo-mfleuti thank you, first this needs to be approved then we can get a design created, the will have a look at this ticket and consider your design.

**Comment by jancborchardt** [comment]
This and all the linked issues asking for a setting, the requests are only for 2 fields:
- Detailed name
- Notes

Notes was also brought up by me some time ago and fixed meanwhile. We can also just add "Detailed name" to show up by default, no config needed.

It should show up as the first block of fields below the heading, as name is of course the most important thing and relates to the info in the heading.
In the long term we can also move over to splitting the fields like Android and iOS contacts do, although then we need to rethink the header approach. FYI @nimishavijay

**Comment by Peque** [comment]
@jancborchardt Please note that in https://github.com/nextcloud/contacts/issues/3385 the request does not seem to specify a particular field of interest, and in https://github.com/nextcloud/contacts/issues/4537 it says "like notes", but it is not clear it means *only* notes. Others have explicitly mentioned other fields [like the website](https://github.com/nextcloud/contacts/issues/1951#issuecomment-1939370343).

Also, some users (like myself) may have up-voted the issue without adding a comment listing other fields since the title and description were already about making this configurable (not just adding a specific field). For example, I'm more interested in "Gender", "phonetics"/"nickname" and "LinkedIn" social network. I have many work-related contacts and those fields are important to me. The first one to use "he/she/them" correctly when you don't know just by the name (i.e.: some names in Chinese can be used by both genders, and names simply in foreign languages can be hard to recognize if you are not used to them).

I'd suggest opening a new issue instead of renaming its title and changing its scope, if adding a detailed name to the default fields is a step you'd like to take.

**Comment by lexo-mfleuti** [comment]
@jancborchardt I disagree. All address fields have to be clearly and properly separated. This also comes into play for example with new laws in Switzerland where address fields have to be clearly separated for billing purposes. For example the field `address` must be separated into `street` + `number`. More details about this here:

https://www.six-group.com/en/products-services/banking-services/payment-standardization/swiss-payments.html#scrollTo=structured-hybrid-address

Since Nextcloud contacts are increasingly more often synced with other business-related software including ERP systems it must provide proper contact data handling.

I'm sorry to say that this is very bad practice:

<img width="696" height="151" alt="Image" src="https://github.com/user-attachments/assets/375670a3-eb66-4e95-8c56-f15111a0b4b1" />

All contact fields must be properly separated as well as clearly and easily visible to the users. Otherwise contact dependent business processes will break.

**Comment by jancborchardt** [comment]
Renamed the issue to previous title. @ChristophWurst my suggestion would be to go with "Detailed name" first as mentioned in https://github.com/nextcloud/contacts/issues/1951#issuecomment-3640916146

@lexo-mfleuti could you open a separate issue about the street + number separation? Address is already quite separated, and this would just be a small added change. If it’s all part of this big issue it’s less likely to happen.
